core/src/main/java/fr/sncf/osrd/conflicts/SpacingResourceGenerator.kt

```diff
@@ -201,35 +179,34 @@ class SpacingRequirementAutomaton(
             val signalOffset = incrementalPath.toTravelledPath(pathSignal.pathOffset)
             val sightOffset = signalOffset - rawInfra.getSignalSightDistance(physicalSignal)
             // If the train's simulation has reached the point where the signal is seen, bail out
-            if (
-                callbacks.currentPathOffset < sightOffset &&
-                    !(callbacks.simulationComplete && incrementalPath.pathComplete)
-            )
-                break@signalLoop
+            if (callbacks.currentPathOffset < sightOffset) break@signalLoop
             val sightTime = callbacks.arrivalTimeInRange(sightOffset, signalOffset)
             assert(sightTime.isFinite())
-            val signalProtectedZone = getSignalProtectedZone(pathSignal)
 
-            if (nextProbedZoneForSignal == -1) nextProbedZoneForSignal = signalProtectedZone
+            val firstBlockIndex = pathSignal.minBlockPathIndex
+            val blocks = mutableStaticIdxArrayListOf(incrementalPath.getBlock(firstBlockIndex))
+            var nSimulatedZones =
+                blockInfra.getBlockPath(blocks[0]).size // number of zones in the blocks
+            val firstZone = incrementalPath.getBlockStartZone(firstBlockIndex)
+            var lastZone = firstZone + 1 // excluded
 
             // find the first zone after the signal which can be occupied without disturbing the
             // train
             zoneProbingLoop@ while (true) {
-                // if we reached the last zone, just quit
-                if (nextProbedZoneForSignal == zoneEndIndex) {
-                    if (incrementalPath.pathComplete) break@zoneProbingLoop
-                    return NotEnoughPath
+                // setup the block / zone arrays
```

Comment:
I'd love a comment as follows:

In the process of figuring out whether a zone's occupancy would disturb the train, we need to simulate more and more blocks along the way.